data/src/command.rs

```diff
@@ -65,10 +65,34 @@ pub enum Command {
     Mode(String, Option<String>, Option<Vec<String>>),
     Away(Option<String>),
     SetName(String),
+    OpenBuffer(String),
     Raw(String),
     Unknown(String, Vec<String>),
 }
 
+impl Command {
+    pub fn is_internal(&self) -> bool {
+        match self {
+            Command::Join(_, _) => false,
+            Command::Motd(_) => false,
+            Command::Nick(_) => false,
+            Command::Quit(_) => false,
+            Command::Msg(_, _) => false,
+            Command::Me(_, _) => false,
+            Command::Whois(_, _) => false,
+            Command::Part(_, _) => false,
+            Command::Topic(_, _) => false,
+            Command::Kick(_, _, _) => false,
+            Command::Mode(_, _, _) => false,
+            Command::Away(_) => false,
+            Command::SetName(_) => false,
+            Command::OpenBuffer(_) => true,
+            Command::Raw(_) => false,
+            Command::Unknown(_, _) => false,
+        }
+    }
+}
```

Comment:
Let's have `Command::Internal` and `command::Internal::OpenBuffer`. New variants can be added there and its type enforced better.

Then `Input::internal` can return `Option<command::Internal>` and caller doesn't have to deal w/ impossible states of dealing w/ non-internal commands during pattern matching